### PR TITLE
emacs{,-app}: update to 28.1

### PR DESCRIPTION
**Pull request by amake**
#### Description

Update to 28.1

I have lightly tested emacs and emacs-app with the default variants.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 12.3.1 21E258 arm64
Xcode 13.3 13E113

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: editors/emacs/Portfile [hunk 1]
```diff
@@ -211,10 +196,14 @@ if {$subport eq $name || $subport eq "emacs-devel"} {
         depends_lib-append       port:dbus
     }
 
+    variant xwidgets requires gtk description {Enable use of xwidgets in Emacs buffers} {
```
Comment (wyuenho) [inline editors/emacs/Portfile:199]: xwidgets does not require GTK. It works with AppKit now.

**Comment by wyuenho** [comment]
These lines need to be replaced with `(set-fontset-font t 'emoji '("Apple Color Emoji" . "iso10646-1") nil 'prepend)`

https://github.com/macports/macports-ports/blob/master/editors/emacs/files/site-start.el#L16-L18

File: editors/emacs/Portfile (hunk 1 above)
Comment (amake) [reply editors/emacs/Portfile:199]: Doesn't it require GTK if you're not building the NS app? The -app subports have a separate variant declaration that doesn't require GTK.

**Comment by amake** [comment]
> These lines need to be replaced with `(set-fontset-font t 'emoji '("Apple Color Emoji" . "iso10646-1") nil 'prepend)`
> 
> https://github.com/macports/macports-ports/blob/master/editors/emacs/files/site-start.el#L16-L18

Thanks, done.

File: editors/emacs/Portfile (hunk 1 above)
Comment (wyuenho) [reply editors/emacs/Portfile:199]: Oh right, my mistake.

**Review by wyuenho — APPROVED** [review]

**Pull request merged**